torchgeo/datasets/geonrw.py

```diff
@@ -0,0 +1,334 @@
+        'dem': lambda path: Image.open(path).copy(),
+        'seg': lambda path: Image.open(path).convert('I;16'),
+    }
+
+    modality_filenames: ClassVar[dict[str, Callable[[list[str]], str]]] = {
+        'rgb': lambda utm_coords: '{}_{}_rgb.jp2'.format(*utm_coords),
+        'dem': lambda utm_coords: '{}_{}_dem.tif'.format(*utm_coords),
+        'seg': lambda utm_coords: '{}_{}_seg.tif'.format(*utm_coords),
+    }
+
+    modalities: ClassVar[list[str]] = ['rgb', 'dem', 'seg']
+
+    url: ClassVar[str] = (
+        'https://huggingface.co/datasets/torchgeo/geonrw/resolve/main/{}'
+    )
+    filename: ClassVar[str] = 'nrw_dataset.tar.gz'
+    md5: ClassVar[str] = 'd56ab50098d5452c33d08ff4e99ce281'
+
+    def __init__(
+        self,
+        root: Path = 'data',
+        split: str = 'train',
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize the GeoNRW dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            DatasetNotFoundError: If dataset is not found and *download* is False.
+        """
+        assert split in self.splits, f'split must be one of {self.splits}'
+
+        self.root = root
+        self.split = split
+        self.transforms = transforms
+        self.download = download
+        self.checksum = checksum
+
+        if split == 'test':
+            self.city_names = self.test_list
+        elif split == 'train':
+            self.city_names = self.train_list
+
+        self._verify()
+
+        self.file_list = self._get_file_list()
+
+    def _get_file_list(self) -> list[str]:
+        """Get a list of files for cities in the dataset split.
+
+        Returns:
+            list of filenames in the dataset split
+        """
+        file_list: list[str] = []
+        for cn in self.city_names:
+            pattern = os.path.join(self.root, cn, '*rgb.jp2')
+            file_list.extend(glob(pattern))
+        return sorted(file_list)
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.file_list)
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        to_tensor = transforms.ToTensor()
```

Comment:
Is this really needed? It's just a 1-liner to use `torch.from_numpy()` right? Eventually I'm guessing they will replace transforms v1 with v2, which will be a breaking API change. Just trying to reduce our reliance on torchvision.

Reply:
it's the standard conversion from PIL images so I just used that. Otherwise, it would be `torch.from_numpy(np.asarray(self.readers[modality](modality_path)))`

Reply:
I guess I'm okay with it, we can always change it later if the transforms API has breaking changes.